sdk/helper/useragent/useragent.go

```diff
@@ -26,22 +26,39 @@ var (
 // String returns the consistent user-agent string for Vault.
 //
 // e.g. Vault/0.10.4 (+https://www.vaultproject.io/; go1.10.1)
-func String() string {
-	return fmt.Sprintf("Vault/%s (+%s; %s)",
-		versionFunc(), projectURL, rt)
+//
+// Given comments will be appended to the semicolon-delimited comment section.
+//
+// e.g. Vault/0.10.4 (+https://www.vaultproject.io/; go1.10.1; comment-0; comment-1)
+func String(comments ...string) string {
+	ua := fmt.Sprintf("Vault/%s (+%s; %s", versionFunc(), projectURL, rt)
+
+	for _, c := range comments {
+		ua = fmt.Sprintf("%s; %s", ua, c)
+	}
```

Comment:
Maybe can use https://pkg.go.dev/strings#Join ?

Reply:
Good idea. Done in ae8f4ba.